lib/js/components/Banner/Banner.vue

```diff
@@ -0,0 +1,375 @@
+					:touchable="false"
+					@click.native="toggleExpandedText"
+				/>
+			</div>
+			<div v-if="!hasExpandedText && closable">
+				<ds-icon-button
+					:size="ICON_BUTTON_SIZES.SMALL"
+					:icon="ICONS.FA_XMARK"
+					:color="ICON_BUTTON_COLORS.NEUTRAL"
+					:radius="BUTTON_RADIUSES.ROUNDED"
+					:touchable="false"
+					@click.native="$emit('close')"
+				/>
+			</div>
+		</div>
+		<div v-if="hasExpandedText && expanded" class="ds-banner__expandedContainer">
+			<ds-divider :prominence="DIVIDER_PROMINENCES.STRONG" />
+			<div class="ds-banner__expandedText">
+				<slot name="expandedText" />
+			</div>
+		</div>
+	</div>
+</template>
+
+<style lang="scss" scoped>
+@import '../../../styles/settings/colors/tokens';
+@import '../../../styles/settings/spacings';
+@import '../../../styles/settings/typography';
+@import '../../../styles/settings/radiuses';
+@import '../../../styles/settings/media-queries';
+
+.ds-banner {
+	$self: &;
+
+	@mixin iconContainerMobileStyles {
+		padding: $space-xxxxs 0;
+	}
+
+	@mixin headerMobileStyles {
+		padding: 0 $space-xxxxs;
+	}
+
+	@mixin expanderMobileStyles {
+		padding: 0 $space-xxxxs;
+	}
+
+	@mixin iconMobileStyles {
+		padding: $space-xxxs;
+	}
+
+	border-radius: $radius-m;
+	border-style: solid;
+	border-width: 1px;
+	display: flex;
+	flex-direction: column;
+	padding: $space-xs;
+
+	&.-vertical {
+		#{$self}__buttonTextHorizontal {
+			display: none !important;
+		}
+
+		#{$self}__buttonTextVertical {
+			display: initial !important;
+		}
+
+		#{$self}__iconContainer {
+			@include iconContainerMobileStyles;
+		}
+
+		#{$self}__header {
+			@include headerMobileStyles;
+		}
+
+		#{$self}__expander {
+			@include expanderMobileStyles;
+		}
+
+		#{$self}__icon {
+			@include iconMobileStyles;
+		}
+	}
+
+	&.-warning {
+		background-color: $color-warning-background;
+		border-color: $color-warning-border;
+	}
+
+	&.-success {
+		background-color: $color-success-background;
+		border-color: $color-success-border;
+	}
+
+	&.-info {
+		background-color: $color-info-background;
+		border-color: $color-info-border;
+	}
+
+	&.-fail {
+		background-color: $color-fail-background;
+		border-color: $color-fail-border;
+	}
+
+	&.-neutral {
+		background-color: $color-neutral-background;
+		border-color: $color-neutral-border-weak;
+	}
+
+	&.-default {
+		background-color: $color-default-background;
+		border-color: $color-neutral-border-weak;
+	}
+
+	&__content {
+		display: flex;
+	}
+
+	&__header {
+		@include headerMobileStyles;
+
+		display: flex;
+		justify-content: space-between;
+		margin-right: $space-xxs;
+		width: 100%;
+
+		@media #{breakpoint-m()} {
+			padding: 0 $space-xxs;
+		}
+	}
+
+	&__title {
+		@include headlineXS;
+		@include textBold;
+	}
+
+	&__defaultText {
+		@include textM;
+	}
+
+	&__text {
+		display: flex;
+		flex-direction: column;
+		justify-content: center;
+		flex-grow: 1;
+		padding: $space-xxxxs 0;
+		margin: 0 $space-s;
+
+		:nth-child(n + 2) {
```

Comment:
I'd usually add a modifier to `&__text`, e.g. `-hasItemsBelow` and add `margin-bottom` there, just to avoid being dependent on the HTML structure. Maybe this is fine too and won't blow up in the future, I'm not pushing 🙂

Reply:
Yeah, that's what I meant by adding `-hasItemsBelow` modifier, only if there is button or defaultText provided. I try to avoid `nth-child` because it makes breaking styles easier. Again, I'm not pushing.

Reply:
you were right. it broke vertical button ripple XD